Update branch to latest qutebrowser/master

adhikasp opened the pull request:
This update and resolve conflict the `pinusc/feature/tree-tabs` branch against the latest `qutebrowser/master`.

I reran the test and it seems to still passes. Except these `pdfjs` related test (which I think caused by some problem on my local setup). On runtime, opening pdf ([example](https://lwn.net/Kernel/LDD3/)) still run fine on my end.
```
FAILED tests/end2end/features/test_downloads_bdd.py::test_downloading_via_pdfjs
FAILED tests/end2end/features/test_qutescheme_bdd.py::test_pdfjs_is_used_for_pdf_files
FAILED tests/end2end/features/test_qutescheme_bdd.py::test_pdfjs_is_not_used_when_disabled
FAILED tests/end2end/features/test_qutescheme_bdd.py::test_downloading_a_pdf_via_pdfjs_button_issue_1214
FAILED tests/unit/utils/test_version.py::TestPDFJSVersion::test_real_file - A...
```

I already use this branch as daily browser for 2 days, and haven't found any bugs.

jyscao committed:
Add: new mode for Auto Search

* when `url.auto_search = 'no-url-scheme'`, URLs
  must contain schemes to be considered valid
* implements feature requested in #5038

ju-sh committed:
Use tmpdir and pathlib

ju-sh committed:
use tmp_path and pathlib for tests/unit/misc

jnphilipp committed:
Add cmd argument for control port.

adhikasp committed:
Merge remote-tracking branch 'pinusc/feature/tree-tabs'

adhikasp commented on qutebrowser/mainwindow/treetabwidget.py:
```diff
@@ -35,8 +35,6 @@ def __init__(self, win_id, parent=None):
         # root of the tab tree, common for all tabs in the window
         self.tree_root = Node(None)
         super().__init__(win_id, parent)
-        self.tabBar().tabMoved.disconnect(  # type: ignore
-            self.update_tab_titles)
```
I remove this line because it cause crash on startup. Sadly I am not sure what this line does, as tab title seems to always show correctly.

arza-zara replied:
It was intentionally added for performance in https://github.com/qutebrowser/qutebrowser/pull/4602/commits/b8d9622aa64324a4c96aed205a67e845d5fb2c11. The crash comes from you missing the other change at https://github.com/qutebrowser/qutebrowser/commit/b8d9622aa64324a4c96aed205a67e845d5fb2c11#diff-7c41aa2423e4249167ffe8cbb8680380L70-R70:
```diff
-       bar.tabMoved.connect(functools.partial(
-           QTimer.singleShot, 0, self.update_tab_titles))
+       bar.tabMoved.connect(self.update_tab_titles)
```

adhikasp replied:
Thanks for the pointer :smile: 

I introduce back the changes.

adhikasp committed:
Disconnect tabMoved signal

Reintroduce change on b8d9622 that was accidentally removed on merge
commit 5ae4982.

nicholaslantz committed:
Add setting to disable back and forward buttons on mouse.

Resolves #5239

kontheocharis committed:
Add smarthide property to hide status only on normal mode

kontheocharis committed:
Make Travis happy

The-Compiler committed:
Fix -1 focus_stack_size in TabDeque

The-Compiler committed:
Update changelog

The-Compiler committed:
pdfjs: Also check for web/viewer.html in is_available()

If a build/pdf.js file exists, but web/viewer.html does not, we would run into
a PDFJSNotFound exception inside of generate_pdfjs_page().

The-Compiler committed:
pdfjs: Move _SYSTEM_PATHS to module-level

This makes it possible to patch them for tests.

nicholaslantz committed:
Changed position of input.mouse_backforward in configdata.

It was out of alphabetic order before.

nicholaslantz committed:
Merge branch 'master' of git://github.com/qutebrowser/qutebrowser

The-Compiler committed:
tests: Stabilize test_qute_settings_persistence

The :jseval command triggers an XHR in JS, and we'll need to wait for that to
actually finish (async) before the setting is really set.

See https://github.com/qutebrowser/qutebrowser/issues/5390#issuecomment-628773737

The-Compiler committed:
Add quirk for missing globalThis

globalThis was introduced in Chrome 71:
https://caniuse.com/#feat=mdn-javascript_builtins_globalthis

Since those websites are quite common and we probably want to continue
supporting Qt 5.12 for a bit, let's bite the bullet and add a small polyfill,
because it's trivial.

Fixes #5486

The-Compiler committed:
Update changelog

pyup-bot committed:
Update certifi from 2020.4.5.1 to 2020.4.5.2

pyup-bot committed:
Update certifi from 2020.4.5.1 to 2020.4.5.2

pyup-bot committed:
Update certifi from 2020.4.5.1 to 2020.4.5.2

pyup-bot committed:
Update certifi from 2020.4.5.1 to 2020.4.5.2

pyup-bot committed:
Update codecov from 2.1.3 to 2.1.4

pyup-bot committed:
Update flake8-comprehensions from 3.2.2 to 3.2.3

pyup-bot committed:
Update mypy from 0.770 to 0.780

pyup-bot committed:
Update lazy-object-proxy from 1.4.3 to 1.5.0

pyup-bot committed:
Update sphinx from 3.0.4 to 3.1.0

pyup-bot committed:
Update pytest from 5.4.2 to 5.4.3

pyup-bot committed:
Update pytest-bdd from 3.3.0 to 3.4.0

pyup-bot committed:
Update pytest-mock from 3.1.0 to 3.1.1

pyup-bot committed:
Update sortedcontainers from 2.1.0 to 2.2.2

pyup-bot committed:
Update wcwidth from 0.2.2 to 0.2.4

pyup-bot committed:
Update tox from 3.15.1 to 3.15.2

The-Compiler committed:
Fix cast for new mypy version

The-Compiler committed:
Hide mypy.ini

kontheocharis committed:
Merge statusbar.hide and statusbar.smarthide options into statusbar.show

kontheocharis committed:
Add statusbar.show to YamlMigrations

kontheocharis committed:
Add additional else clause for statusbar.show=in_mode

The-Compiler committed:
tests: Add ignores for Qt 5.15 Chromium debug build

The-Compiler committed:
Merge remote-tracking branch 'origin/pr/5492'

The-Compiler committed:
Update changelog

The-Compiler committed:
Merge remote-tracking branch 'origin/pr/5483'

kontheocharis committed:
Update naming in qutebrowser/config/configdata.yml

Co-authored-by: Florian Bruhin <me@the-compiler.org>

kontheocharis committed:
Modify modeman.instance() to return specific error when unavailable.

This avoids the need to have a generic 'except KeyError' in
statusbar/bar.py when querying the current mode, in the case when the
mode manager has not been initialized yet.

kontheocharis committed:
Add unreachable test in statusbar/bar.py

kontheocharis committed:
Merge branch 'smart-hide-status' of https://github.com/kontheocharis/qutebrowser into smart-hide-status

The-Compiler committed:
Merge remote-tracking branch 'origin/pr/5445'

The-Compiler committed:
Simplify configdata.yml

The-Compiler committed:
Update docs

kontheocharis committed:
Change in_mode to in-mode everywhere

The-Compiler committed:
Fix ~ with :config-write-py

df4a011d48fa9f12fccf59ebca0954ccdf646d20 introduced a bug because the
os.path.isabs check was done before os.path.expanduser - and isabs('~/...')
will always be False.

The-Compiler committed:
Update changelog

The-Compiler committed:
Add note to generated config.py

The-Compiler committed:
Add arguments for :report

kontheocharis committed:
Add ability to customise tab font based on selected status

Also, use Qt Style Sheets to set the tab bar font instead of
setFont(QFont). This makes the fonts.tabs.* settings consistent with all
the other font settings (using Font instead of QFont).

kontheocharis committed:
Add test for _migrate_multiple

kontheocharis committed:
Remove unnecessary call to update()

Also, split a line that pylint complains is too long.

kontheocharis committed:
Update qutebrowser/config/configdata.yml

Co-authored-by: Marcel Schilling <marcel.schilling@uni-luebeck.de>

The-Compiler committed:
Merge remote-tracking branch 'origin/pr/5512'

The-Compiler committed:
Update docs

The-Compiler committed:
Rename test_migrate_multiple

The-Compiler committed:
Merge remote-tracking branch 'origin/pr/5498'

The-Compiler committed:
Update docs

The-Compiler committed:
Merge remote-tracking branch 'origin/pr/5487'

The-Compiler committed:
tests: Add testutils.change_cwd()

The-Compiler committed:
tests: Fix test_configinit after #5512

The-Compiler committed:
mypy: Fix type for _migrate_to_multiple

The-Compiler committed:
Merge remote-tracking branch 'origin/pr/5485'

The-Compiler committed:
tests: Cleanups after pathlib migration in tests/unit/scripts/

The-Compiler committed:
tests: Remove unused import

The-Compiler committed:
Use debug logging for log config messages

See #5445

kontheocharis committed:
Remove QtFont as a config type

kontheocharis committed:
Remove unneeded imports

kontheocharis committed:
Simplify some tests

kontheocharis committed:
Simplify test_configtypes.py

kontheocharis committed:
Move stylesheet to ConsoleWidget to avoid duplication

kontheocharis committed:
Remove unnecessary argument in TestFont.klass

pierg75 committed:
Added instructions on how to install the Fedora debuginfo packages

Signed-off-by: Pierguido Lambri <plambri@redhat.com>

pierg75 committed:
Fixed a wrong formatting

Signed-off-by: Pierguido Lambri <plambri@redhat.com>

pierg75 committed:
And fixed with the right command

Signed-off-by: Pierguido Lambri <plambri@redhat.com>

pierg75 committed:
Removed the sudo command

Signed-off-by: Pierguido Lambri <plambri@redhat.com>

pierg75 committed:
Fixed the debuginfo command, removed the sudo command and installed the right debug symbol packages for qtwebengine

Signed-off-by: Pierguido Lambri <plambri@redhat.com>

pierg75 committed:
Removed the useless debuginfo suffix

Signed-off-by: Pierguido Lambri <plambri@redhat.com>

The-Compiler committed:
Merge remote-tracking branch 'origin/pr/5515'

The-Compiler committed:
Merge remote-tracking branch 'origin/pr/5516'

The-Compiler committed:
Update changelog

The-Compiler committed:
Merge remote-tracking branch 'origin/pr/5460'

The-Compiler committed:
Update docs

rcorre committed:
Enable configuring loglevel at runtime.

Now `:set logging.level.{ram,console}` will work after qutebrowser has
started. This allows us to derecate debug-set-loglevel.

Logging is initialized too early for config.change_filter to work, so I
had to hack it.

The-Compiler committed:
Remove :debug-log-level

It's a hidden :debug-* command anyways, so no need to keep it around.

The-Compiler committed:
tests: Separate init_from_config logs in utils.test_log

The-Compiler committed:
Update changelog

The-Compiler committed:
Merge remote-tracking branch 'origin/pr/5474'

The-Compiler committed:
Switch some webview/misc logging messages to network logger

The-Compiler committed:
qutescheme: Unify code between qute_log and qute_plainlog

The-Compiler committed:
Merge qute://plainlog into qute://log?plain

The-Compiler committed:
Revert "tests: Separate init_from_config logs in utils.test_log"

Whoops... It uses fixtures from that test class!

This reverts commit a9d10aee8403fdfcfc68d38a40312c7811d0dfe7.

The-Compiler committed:
Fix lint

The-Compiler committed:
Change logging format when the loglevel is changed via config

The-Compiler committed:
Refactor log.LogFilter

This essentially does two things:

1) Use a set for LogFilter

This means we don't support filters like "eggs.bacon" anymore (but we *do*
support loggers like that) - however, those were already disallowed by the
--logfilter argument validation anyways!

In return, we get probably slightly better performance (checking set membership
rather than iterating all filters) and more straightforward code.

2) Move parsing from various locations around the code to the LogFilter class.

The-Compiler committed:
Fix typo

(cherry picked from commit 7dceedb26bd5d0df97a97cd727e19fd9937bd850)

The-Compiler committed:
Add categories to qute://log

The-Compiler committed:
Fix lint

The-Compiler committed:
commands: Add some additional tests for *args handling

(cherry picked from commit 3983a6d3ee30f86b1afd809bc786322cbff8fd55)

The-Compiler committed:
command: Use consistent name for arg_info

(cherry picked from commit 279d28f982f3afe32c94fc938677bcad816b647e)

The-Compiler committed:
Update changelog

The-Compiler committed:
Fix lint

The-Compiler committed:
Fix unhandled SelectionUnsupportedError in :completion-item-yank

The-Compiler committed:
requirements: Remove lxml from dev requirements

d805e2d71e8e54428aae9a814a898e4ba7004118 removed lxml in check_coverage.py,
5e64aae70be25e8871846af15ff92cfce7a816d9 removed ua_fetch.py

Thus, no lxml usages are remaining.

The-Compiler committed:
Improve some init logging

The-Compiler committed:
Adjust version number in session warning

The-Compiler committed:
configdata: Remove comment about darkModeClassifierType

My analysis was correct and the setting now got removed from Chromium:
https://chromium-review.googlesource.com/c/chromium/src/+/2169489

Also see some other refactorings in the linked bug report:
https://bugs.chromium.org/p/chromium/issues/detail?id=1091095

See #5394

The-Compiler committed:
Improve lost focusproxy handling

Disable the workaround if not on Qt 5.11, don't log if nothing was found, and
clean up the code a bit.

The-Compiler committed:
Fix lint

The-Compiler committed:
Add note about colors.webpage.darkmode.policy.images on Qt 5.15.0

See #5505

The-Compiler committed:
Fix exception with invalid feature permission URLs

The-Compiler committed:
Create codeql-analysis.yml

The-Compiler committed:
Code scanning: Exclude C++

nicholaslantz committed:
Merge branch 'master' of git://github.com/qutebrowser/qutebrowser

nicholaslantz committed:
Renamed a couple of mouse settings.

Changed:
- input.mouse_backforward to input.mouse.back_forward_buttons
- input.rocker_gestures to input.mouse.rocker_gestures

The-Compiler committed:
Rename urlmarks.py to test_urlmarks.py

The-Compiler committed:
Improve QtWebKit/QtWebEngine import checks

This handles two cases found in crash reports:

- "AssertionError: qWebKitVersion is not None" when QtWebKit itself is
  importable but qWebKitVersion is not. Fixes #5484.

- "ValueError: PyCapsule_GetPointer called with incorrect name" when importing
  QtWebKit - no idea what's going on here exactly.

Note that there might be other cases in the code where those imports are
attempted and only ImportError is handled (because ValueError really shouldn't
happen there...) - but we can't really track them down until we see more crash
reports with the same issue.

The-Compiler committed:
tests: Stabilize test_qute_settings_persistence for real

Before, the "Config option changed: *" triggered on:

  Config option changed: content.headers.user_agent = ...

https://github.com/qutebrowser/qutebrowser/issues/5390#issuecomment-628773737

The-Compiler committed:
Merge remote-tracking branch 'origin/pr/5502'

The-Compiler committed:
Update tests/manual/mouse.html

The-Compiler committed:
Update changelog

The-Compiler committed:
Fix test_urlmarks.py

The-Compiler committed:
userscripts: Clear self._filepath on OSError

As an additional precaution that a "tainted" path isn't being used anywhere.
See https://github.com/qutebrowser/qutebrowser/security/code-scanning/2

The-Compiler committed:
Use 0600 mode for uerscript FIFO

os.mkfifo uses mode 0666 masked with the umask, i.e. 0644 on most systems by
default. We'd like things to be more restrictive, to make sure other users
can't get any data from another users' userscript FIFO.

This shouldn't make any difference in practice because the FIFO is in
standarddir.runtime() which must have 0700 permissions (and qutebrowser creates
all standarddir directories with 0700 permissions if they don't exist).

However, it's still a good idea to restrict the permissions as a second line of
defense.

The-Compiler committed:
Point to GitHub discussions feature

The-Compiler committed:
basekeyparser: Remove old config format comment

The-Compiler committed:
modeparsers: Refactor to avoid subclassing

Before the changes in this commit, we've had to have a subclassed parser for
every mode, even if there was no special key handling going on in that mode.

With a couple of changes, we can avoid many of those subclasses and only have
subclasses for bigger changes (like hint or register modes).

- The awkward handling of self._modename in _read_config() is now removed.
  _read_config() doesn't take an argument, always uses the mode in self._mode
  and gets called from __init__.
- BaseKeyParser takes the mode as an argument to __init__.
- The class attributes (do_log/passthrough/supports_count) now also get passed
  via the constructor.

The-Compiler committed:
basekeyparser: Improve debug logging

The-Compiler committed:
modeparsers: Refactor how hints are handled

Instead of binding hints to fake :follow-hint commands, we now use a separate
CommandKeyParser and ask that for its match result.

If the key matches with the command parser, it is bound in hint mode, so we
clear the hint keystring and defer to the command handling instead.

If it doesn't, we continue hint handling as usual - however, the HintKeyParser
is now not a CommandKeyParser anymore, so we don't have to deal with command
parsing (and have a custom execute implementation instead).

Closes #4504
Fixes #4392
Fixes #4368
Helps with #5084, though it doesn't completely fix that yet.
Supersedes #3742 (fix for #3735)
Supersedes #4691 (fix for #4264)

The-Compiler committed:
Add BindingTrie.__str__

The-Compiler committed:
Update changelog

The-Compiler committed:
tests: Get rid of fake_keyevent

Use keyutils.KeyInfo(...).to_event() instead.

The-Compiler committed:
Show partial keystrings for all modes in statusbar

Closes #2817

The-Compiler committed:
Handle OSErrors gracefully in os.utime()

For some reason, some systems remove the IPC socket of qutebrowser, causing it
to die after 6h. Now we handle this case more gracefully and also make sure it
still exists all 5 minutes.

Fixes #996 (finally!), though I still have no idea *why* it is happening...

The-Compiler committed:
Fix lint

The-Compiler committed:
Add git branch to :version

The-Compiler committed:
Fix tests/lint

jyscao committed:
Rename new auto_search mode for less ambiguity

* no-url-scheme -> schemeless
* improve message to log.url.debug

jyscao committed:
Add tests for new auto_search mode

* create UrlParams cls to hold default param values for use in test_is_url
* reparameterize test_is_url using instances of UrlParams
* refactor test_is_url pytest function to reflect changes:
  * add new conditional to test new auto_search mode (schemeless)
  * add new param (is_url_in_schemeless) to test schemeless mode
  * revise docstring to group attrs of UrlParams & include new param above

jyscao committed:
Add check for space character for schemeless mode

* this catches the edge case of urlstr w/ valid schemes, but
  contain space(s), such urlstr should be treated as non-URLs
* e.g. "site:cookies.com oatmeal raisin" should be auto-searched

jyscao committed:
Fix pylint errors (line-too-long)

* abbreviate some comment strings; e.g. because -> b/c, space -> spc
* rename parameter: uses_dns -> use_dns (shorter identifier)

jyscao committed:
Break long conditional over 2 lines (fix pylint error)

jyscao committed:
Re-add accidentally replaced comment on test case

jyscao committed:
Document defaults of the 4 optional UrlParams attrs

jyscao committed:
One last pylint fix

The-Compiler committed:
Add workaround for muted indicator with Qt 5.15

See #5525

The-Compiler committed:
Re-enable content.dns_prefetch by default

This reverts aefdf7565a9b16ec29d3461e7afaadb4afe86f0a - see #4657

The-Compiler committed:
Work around segfaults when updating QWebEngineProfile HTTP headers

With Qt versions between 5.12 and 5.14 (inclusive), calling setHttpUserAgent()
or setHttpAcceptLanguage() can cause segfaults on subsequent loads:
https://bugreports.qt.io/browse/QTBUG-75884

The easiest way to reproduce this is by using WhatsApp Web and reloading after
setting the user-agent. While the bug should've been fixed in Qt 5.12.5, 5.13.1
and 5.14.0, that's not actually the case - only Qt 5.15 fixed it properly.

Since we use a request interceptor to set the headers anyways, just don't call
those problematic methods at all when updating settings at runtime.

Fixes #4657

The-Compiler committed:
Merge branch 'master' into docked-inspector

The-Compiler committed:
mypy: Remove disallow_incomplete_defs

Removed from other sections in 3679d3a05b273cbac206c52632ae0c369e7cccf2 but
those were added in the docked-inspector branch.

The-Compiler committed:
miscwidgets: Add parent argument to InspectorSplitter

The-Compiler committed:
miscwidgets: Fix bare except for InspectorSplitter

The-Compiler committed:
miscwidgets: Use class attributes for InspectorSplitter constants

The-Compiler committed:
miscwidgets: Distribute handle width evenly in InspectorSplitter

This makes the behavior (and thus tests) more uniform, because in the other
cases, the splitter width is evenly distributed as well.

The-Compiler committed:
miscwidgets: Add tests for InspectorSplitter

The "minimal window size" one is still failing with top/right for some
reason...

The-Compiler committed:
miscwidgets: Clean up _adjust_size

Should still do the same, just be a bit more readable.

The-Compiler committed:
miscwidgets: Remove setStretchFactor for InspectorSplitter

For some reason, if the stretch factor is set, the "very small window" handling
is wrong with the inspector being at top/left: Qt decides to hide the inspector
outright with small window sizes.

The stretch factor don't seem to affect anything else (since we're handling
most sizing manually anyways), so let's just remove it.

The-Compiler committed:
Fix lint

The-Compiler committed:
Update docs

The-Compiler committed:
Set window title for inspector

Closes #5451

The-Compiler committed:
tests: Generalize configinit tests

The-Compiler committed:
Add overlay scrollbar

See #2377

The-Compiler committed:
tests: Fix JS header tests

This is a follow-up to f0eed465ab608061212366c6ebc26a0c47803060
See #4657

The-Compiler committed:
Merge branch 'master' into docked-inspector

The-Compiler committed:
tests: Improve InspectorSplitter sizing tests

- Use a smaller size (600 instead of 1000) as base so that the test works with
  smaller screens (e.g. with macOS on Travis, the screen resolution is
  1024x768, with 658px being available vertically).
- Skip the test instead of failing when resizing doesn't work (e.g. when
  running under a tiling WM without Xvfb)
- Fix handle calculation if the handle has an odd number of pixels (e.g. 5px on
  macOS).

The-Compiler committed:
inspector: Refactor how opening/closing work

The inspector object is now re-used and only created once (when the inspector
is opened first). This also moves some logic from CommandDispatcher to the
inspector object.

The geometry is now saved again when the window is closed (via closeEvent)
rather than in AbstractWebInspector.set_position - after all, the window will
also close when the position is changed.

Finally, set_position(None) changed its meaning: Its now not used anymore to
close/delete the inspector, but instead means "use the last saved position".

The-Compiler committed:
Use separate inspector section in state config

The-Compiler committed:
Add AbstractTabPrivate.toggle_inspector

The-Compiler committed:
Shut down inspector when associated tab is closed

Fixes #4348

adhikasp committed:
Merge remote-tracking branch 'origin/master' into feature/tree-tabs

The-Compiler committed:
inspector: Detach when inspector is hidden

ju-sh committed:
Fix 3.5 incompatibility in script

ju-sh committed:
Revert "fix error in asciidoc2html.py script"

This reverts commit 64ffce27f9348e14836528c0313d3048669a29c7.

The-Compiler committed:
inspector: Make detaching to a window work properly

ju-sh committed:
use Path.read_text

The-Compiler committed:
README: Update list of similar projects

d-xo committed:
faq: clarify secion on noscript bindings

The-Compiler committed:
Remove old :inspect note

This was true for QtWebKit and doesn't apply for QtWebEngine anymore.

See #730

The-Compiler committed:
Remove Qt 5.8 workarounds for _remove_tab

Qt 5.8 support was removed in April 2018 with
69a58c959705b2dd469ba416250f884472593bee, and the last crash report I got with
Qt 5.8 was in the same month.

The workaround removed in this commit still were around, and despite of what
the version check says, they are *not* needed with Qt 5.7.

Those were workarounds for the following bugs:

- https://bugreports.qt.io/browse/QTBUG-58698
  (#2290, 1af951fd62b4878b2eaeb63e628b0ce5c06eeae8)

- https://bugreports.qt.io/browse/QTBUG-58982
  (#2261, 90f472bf590b6fdb3e7f4158ae420002ccbdee17)

See #3608

The-Compiler committed:
miscwidgets: Support proper unwrapping in WrapperLayout

Don't only reparent the widget, also make sure we don't actually control (and
e.g. resize) it anymore.

The-Compiler committed:
miscwidget: Base WrapperLayout on QVBoxLayout

This way we don't have to care about things like sizeHint manually.

The-Compiler committed:
Disable PseudoLayout workaround except on Qt 5.11.0

The-Compiler committed:
Move inspector command to components.misccommands

The-Compiler committed:
Fix mypy

The-Compiler committed:
Rename :inspector to :devtools

The general consensus seems to be that the inspector is the first tab of the
devtools, and Chromium also calls them devtools everywhere.

Closes #5438

The-Compiler committed:
Show error when devtools resources are unavailable on Fedora

Closes #5501

The-Compiler committed:
Split WebEngineInspector into two classes

This way we don't need to have if's everywhere, and the two classes only have
some 3 lines in common anyways!

The-Compiler committed:
inspector: Split _show_window()/_show_docked() from set_position()

The-Compiler committed:
Add completion for inspector positions

The-Compiler committed:
completion: Make column sizing work with one column only

The-Compiler committed:
Simplify AbstractWebInspector._load_position

The-Compiler committed:
Fix AbstractWebInspector docstring

The-Compiler committed:
tests: Move some fixtures to tests/helpers/fixtures.py

The-Compiler committed:
Make it possible to call WrapperLayout.unwrap() before wrapping

The-Compiler committed:
Remove inspector detaching again

It doesn't work properly: If we detach, we'd also need to re-attach when
reopening the inspector!

Let's just not do it for now and see what people think. That also has some
benefits like the inspector remembering what tab the user was on when closing
it.

The-Compiler committed:
Add some tests for AbstractInspector

The-Compiler committed:
Fix lint

The-Compiler committed:
Revert "miscwidget: Base WrapperLayout on QVBoxLayout"

This reverts commit 69a225bcf2a62207bb13d6c2d7c9cd1726146909.

Fails tests and ends up being the more complex solution.

The-Compiler committed:
inspector: Simplify showing in a window

The-Compiler committed:
Add additional devtools keybindings

The-Compiler committed:
Work around devtools window detaching bug

Recreate the inspector when it was shown inline and gets detached to work
around an issue where QtWebEngine (for some odd reason) doesn't show a window
at all.

See #5540

The-Compiler committed:
Remove now unneeded WrapperLayout.unwrap

The-Compiler committed:
Merge branch 'docked-inspector'

The-Compiler committed:
Update changelog

The-Compiler committed:
Add ASCII-art logo to :version

Thanks to Ingvix in IRC for the ASCII-art! :)

The-Compiler committed:
Upgrade requirements

    Upgrade PyVirtualDisplay
    Update virtualenv from 20.0.21 to 20.0.24
    Update pytest-xvfb from 1.2.0 to 2.0.0
    Update pytest-instafail from 0.4.1.post0 to 0.4.2
    Update pytest-cov from 2.9.0 to 2.10.0
    Update py-cpuinfo from 5.0.0 to 6.0.0
    Update py from 1.8.1 to 1.8.2
    Update more-itertools from 8.3.0 to 8.4.0
    Update hypothesis from 5.16.0 to 5.18.0
    Update sphinx from 3.1.0 to 3.1.1
    Update setuptools from 47.1.1 to 47.3.1
    Update mypy from 0.780 to 0.781
    Update pep8-naming from 0.10.0 to 0.11.1
    Update flake8 from 3.8.2 to 3.8.3
    Update requests from 2.23.0 to 2.24.0
    Update codecov from 2.1.4 to 2.1.7
    Update certifi from 2020.4.5.2 to 2020.6.20

The-Compiler committed:
Regenerate docs

The-Compiler committed:
Merge remote-tracking branch 'origin/pr/5536'

The-Compiler committed:
Edit FAQ changes

The-Compiler committed:
Merge remote-tracking branch 'origin/pr/5153'

The-Compiler committed:
Update docs

The-Compiler committed:
Fix test_version

Needs re.MULTILINE because of the qutebrowser ASCII art logo...

The-Compiler committed:
Revert "Remove now unneeded WrapperLayout.unwrap"

This reverts commit 370bd12a1512f5164be0e3ae1838515957e587fb.

Turns one of out the workarounds removed in
87d7dd93420ab92a1a209919297371dc0fadcecd is still needed until Qt 5.12...

The-Compiler committed:
Re-add segfault workarounds

Some tests (e.g. in test_invocations.py) still seem to segfault without this.
This partially reverts 87d7dd93420ab92a1a209919297371dc0fadcecd.

See #2261

The-Compiler committed:
Fix handling of odd splitter size in InspectorSplitter tests

Turns out 7a6e4821a1d5e1cd472e1be639c70858adec93d8 was the wrong way around!

The-Compiler committed:
Use :devtools in inspector test

The-Compiler committed:
Delete widget in WrapperLayout.unwrap()

This was removed in f5f3bf63b58d5a2a73a777b199b9dbc0b360fb66 but is still
needed so that the underlying QWebEngineView is actually deleted with Qt <
5.12. Otherwise, test_spawning_an_editor_and_closing_the_tab in
test_editor_bdd.py would fail because the element doesn't actually vanish.

The-Compiler committed:
tests: Set splitter handle width for InspectorSplitter tests

How the handle width of 5 is distributed on macOS seems like totally random.
Instead of trying to get our calculations to work, just force the handle width
to an even number.

The-Compiler committed:
Revert "Remove Qt 5.8 workarounds for _remove_tab"

After 86d794320343e58467d34daad8cc9a1b8011a512, this now also re-adds the
workaround for QTBUG-58982 (#2290) because it is apparently still needed with
Qt 5.7...

This reverts commit 87d7dd93420ab92a1a209919297371dc0fadcecd.

adhikasp committed:
Merge branch 'master' into feature/tree-tabs

ju-sh committed:
redo old edit

The-Compiler committed:
tests: Fix flakyness in editor bdd tests

There likely was a race condition between JS processing the input and us
clicking the button there. Log the text async from JS via oninput= to avoid
having to click a button.

See https://github.com/qutebrowser/qutebrowser/issues/5390#issuecomment-625179233

The-Compiler committed:
Merge remote-tracking branch 'origin/pr/5534'

The-Compiler committed:
tests: Simplify some open() calls

The-Compiler committed:
Re-enable "Running :spawn with invalid quoting"

Let's assume this doesn't segfault anymore nowadays and let's see what happens.

Closes #1614

The-Compiler committed:
mypy requirements: Add lxml

The-Compiler committed:
Add "tox -e mypy-diff"

Not done by default with "tox -e mypy" because it disables mypy's caching,
causing its runtime to go from <1s to ~15s:
https://github.com/python/mypy/issues/9041

See #1456

The-Compiler committed:
Update PyPI classifiers

The-Compiler committed:
ci: Switch to qutebrowser/ci dockerfiles

The-Compiler committed:
ci: Switch from TRAVIS to CI env vars

zorbulator committed:
add transparency

The-Compiler committed:
tests: Stabilize test_open_with_ascii_locale

The loaded page caused a 404, but we didn't wait for the new error logging
message. Most of the time that went well because we quit before it could be
logged, but not always.

See https://github.com/qutebrowser/qutebrowser/issues/5390#issuecomment-629268747

The-Compiler committed:
Fix FAQ formatting

The-Compiler committed:
Change permission workaround logger to debug

Now that we know how this happens (notification permissions in incognito mode),
we don't need to log a warning here.

The-Compiler committed:
Update user agent completions

The-Compiler committed:
Edit changelog

The-Compiler committed:
Update version in session warning

The-Compiler committed:
Release v1.13.0

The-Compiler committed:
asciidoc2html: Resolve path for sys.path

For some reason, __file__ was "scripts/dev/../asciidoc2html.py", causing
.parents[1] to point to "scripts/dev/" instead of "scripts/".

See #5534

The-Compiler committed:
manifest: Prune .mypy_cache

The-Compiler committed:
Fix duplicate import

The-Compiler committed:
Improve auto_save.session docs

See #4724

The-Compiler committed:
Merge remote-tracking branch 'origin/pr/5546'

The-Compiler committed:
Update changelog

adhikasp committed:
Merge branch 'master' into feature/tree-tabs
